storage/netstore.go

```diff
@@ -234,20 +233,20 @@ func (n *NetStore) RemoteFetch(ctx context.Context, req *Request, fi *Fetcher) e
 			"remote.fetch")
 		osp.LogFields(olog.String("ref", ref.String()))
 
+		ctx = context.WithValue(ctx, "remote.fetch", osp)
+
 		log.Trace("remote.fetch", "ref", ref)
 
 		currentPeer, err := n.RemoteGet(ctx, req, n.localID)
 		if err != nil {
 			log.Trace(err.Error(), "ref", ref)
 			osp.LogFields(olog.String("err", err.Error()))
 			osp.Finish()
-			return ErrNoSuitablePeer
+		} else {
```

Comment:
can you explain why is this change needed?

Reply:
that PR has stagnated and should be closed since there was not resolution on it. I'm not sure why this change is being piggybacked on this PR. are the unit tests and integration tests failing without this change?